### ISSUE TITLE
Allow to use macro calls in enum definitions

### ISSUE THREAD

**Issue by jhass** [crystal-lang/crystal#424]
Feature request:

```ruby
macro make_foo x
  {{x}}_u32 * 3
end

enum Foo
  A = make_foo 0
  B = make_foo 1
end
```

Bonus: allow the macro definition inside the enum.

**Comment by jhass** [comment]
Related to #429 and #236

**Comment by asterite** [comment]
I can probably fix this for 0.18.0 too
